app/src/main/java/org/simple/clinic/editpatient/EditPatientUpdate.kt

```diff
@@ -51,7 +52,7 @@ class EditPatientUpdate(
       is ZoneChanged -> next(model.updateZone(event.zone))
       is StreetAddressChanged -> next(model.updateStreetAddress(event.streetAddress))
       is BackClicked -> onBackClicked(model)
-      is PatientSaved -> dispatch(GoBackEffect)
+      is PatientSaved -> next(model.buttonStateChanged(ButtonState.SAVED), GoBackEffect)
```

Comment:
This screen does not have a valid "Saved" state since we close the screen when that happens. The model should reflect that.

I think we can introduce a new enum here with two states:

- `SavingPatient`
- `NotSavingPatient`

And use these to represent the state.

Reply:
`NotSavingPatient` doesn't make sense to me, what do you think? 

States 
* `SavingPatient` concludes button state in-progress
* `SavedPatient` concludes success and button state can now be enabled

Reply:
The problem is that `SavedPatient` is not a good name for a state, but a name for an event. We need a name that indicates this.